artemis/blocklist.py

```diff
@@ -91,10 +94,16 @@ def should_block_scanning(
         if item.mode != BlocklistMode.BLOCK_SCANNING_AND_REPORTING:
             continue
 
-        if item.domain:
+        if item.domain_and_subdomains:
             if not domain:
                 continue
-            if not is_subdomain(domain, item.domain):
+            if not is_subdomain(domain, item.domain_and_subdomains):
+                continue
+
+        if item.subdomains:
+            if not domain:
```

Comment:
you can flatten 3 if statements
`if not domain`
`if item.domain_and_subdomains and not is_subdomain(...)`
`if item.subdomains and not is_subdomain(...)`